Tests/UnitTests/TestUnits.py

```diff
@@ -0,0 +1,196 @@
+# **************************************************************************
+#
+# MDANSE: Molecular Dynamics Analysis for Neutron Scattering Experiments
+#
+# @file      Tests/UnitTests/TestGeometry.py
+# @brief     Implements module/class/test TestGeometry
+#
+# @homepage  https://mdanse.org
+# @license   GNU General Public License v3 or higher (see LICENSE)
+# @copyright Institut Laue Langevin 2013-now
+# @copyright ISIS Neutron and Muon Source, STFC, UKRI 2021-now
+# @authors   Scientific Computing Group at ILL (see AUTHORS)
+#
+# **************************************************************************
+
+#MDANSE : Molecular Dynamics Analysis for Neutron Scattering Experiments
+#------------------------------------------------------------------------------------------
+#Copyright (C)
+#2015- Eric C. Pellegrini Institut Laue-Langevin
+#BP 156
+#6, rue Jules Horowitz
+#38042 Grenoble Cedex 9
+#France
+#pellegrini[at]ill.fr
+#goret[at]ill.fr
+#aoun[at]ill.fr
+#
+#This library is free software; you can redistribute it and/or
+#modify it under the terms of the GNU Lesser General Public
+#License as published by the Free Software Foundation; either
+#version 2.1 of the License, or (at your option) any later version.
+#
+#This library is distributed in the hope that it will be useful,
+#but WITHOUT ANY WARRANTY; without even the implied warranty of
+#MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#Lesser General Public License for more details.
+#
+#You should have received a copy of the GNU Lesser General Public
+#License along with this library; if not, write to the Free Software
+#Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+
+''' 
+Created on May 29, 2015
+
+@author: Eric C. Pellegrini
+'''
+
+import unittest
+
+from MDANSE.Framework.Units import UnitError, measure
+
+class TestUnits(unittest.TestCase):
+    '''
+    Unittest for the geometry-related functions
+    '''
+        
+    def test_basic_units(self):
+
+        m = measure(1.0,'kg')
+        self.assertAlmostEqual(m.toval('g'),1.0e+03,10)
+        self.assertAlmostEqual(m.toval('mg'),1.0e+06,10)
```

Comment:
This is not good unit test design, since if the first assertion fails, then all the following ones within this massive method will be skipped. Ideally, each assertion statement should be in a separate method, though in this case I think having these two assertions related to kg within one method would be acceptable. Nevertheless, each `measure()` call afterwards should be in a separate method.

Reply:
sorry for not being a good unit test designer !

Reply:
I am in partial agreement with you. Indeed, that unit test is massive but its goal is to test the 9 basic units. For me, breaking into 9 unit tests is not that good design too.  Anyway, I will take your remark into consideration.

Reply:
https://stackoverflow.com/questions/762512/is-it-bad-practice-to-have-more-than-one-assertion-in-a-unit-test

Reply:
I think that the truth is in-between. For instance if I test the prefixes. I think it is reasonable to build one unit test for all prefix and not a unit test per prefix.

Reply:
Firstly, apologies if I worded that initial comment incorrectly; I was trying to point out a place where the code can be improved and nothing else. 

If testing all prefixes is the goal, [parametrisation](https://stackoverflow.com/questions/32899/how-do-you-generate-dynamic-parameterized-unit-tests-in-python) is the way to go, I would say. But, as I have belatedly noticed, this method tests the units and prefixes, so now that I think about it, it would be good to separate these aspects.

Further, I used this method as an example, but in the initial comment, I was commenting on all the test cases, especially the later ones. For example, `test_product_units()` uses three assertions, where each assertion relies on the previous one.

Would it be ok if I rewrote these tests?